components/UnifiedDocFeed/utils/handleFilterSelect.ts

```diff
@@ -10,6 +10,27 @@ type Args = {
   timeScope?: string;
 };
 
+const _getTagsValue = ({ incomingTags, query }) => {
+  const isTagsAString = typeof query.tags === "string";
+  const isTagsAnArray = Array.isArray(query.tags);
+
+  const existingTags = isTagsAString ? [query.tags] : isTagsAnArray ? [...query.tags] : [];
+  
+  let newTags:string[] = [...existingTags]
+  if (incomingTags && incomingTags.length > 0) {
+    for (let i = 0; i < incomingTags.length; i++) {
+      const tagAlreadyInList = existingTags.includes(incomingTags[i]);
+      if (tagAlreadyInList) {
+        newTags = newTags.filter((t) => t !== incomingTags[i]);
+      } else {
+        newTags.push(incomingTags[i]);
+      }
+    }
+  }
+
+  return newTags;
+}
+
```

Comment:
I have a feeling that this can be simplified. But it's kind of hard to follow what's happening since I 'm not sure what the entire structure is like with tags. I'll leave it up to you.